extract/test/test_bbox.py

```diff
@@ -11,7 +11,7 @@ class TestBboxClass(BBoxMixin):
 def test_bbox_mixin():
     t = TestBboxClass()
     bb = t.bbox
-    assert bb.x0 == 459.84
-    assert bb.y0 == 753.697
-    assert bb.x1 == 462.075
-    assert bb.y1 == 765.21
+    assert bb.upper_left_coordinate.x == 459.84
```

Comment:
I'm not really a fan of using asserts for testing.

As soon as one assert fails, code execution stops, so you find at most one problem at a time.

Also, you're testing 4 things here. It might make sense to lump these in as a single test, you're really testing that all the coordinates got read into the right place. But if this fails, you're going to get told that `test_bbox_mixin` failed, and it isn't really clear why or how it failed.

I'd suggest taking a look at the `UnitTest` class built in to Python. Something like https://realpython.com/python-testing/#writing-your-first-test might be useful to read.

I think you'd end up with something like:

````python
class Test_BBox_Coordinates(unittest.TestCase):
    def setUp()
        self.t = TestBboxClass()
        self.bb = t.bbox

    def test_upper_left_x(self):
        self.assertEqual(self.bb.upper_left_coordinate.x, 459.84)

    def test_upper_left_y(self):
        self.assertEqual(self.bb.upper_left_coordinate.y, 753.697)
````

That way, you're going to get very specific errors telling you that lower_right_coordinate.y is broken, but everything else is fine; or that both y coordinates are broken but the xes are fine... or you'll get unexpected errors because bb.upper_left_coordinate is a NoneType and so on. Whichever bit fails, all of the tests will be run, and you'll be able to see all the things that failed, not just the first.